llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -411,6 +417,21 @@ std::string saveModuleIR(Module &M, int I, StringRef Suff) {
   return OutFilename;
 }
 
+bool isImportedFunction(const Function &F) {
+  // Functions with definitions are not imported
+  if (!F.isDeclaration())
+    return false;
```

Comment:
We should add an early exit for LLVM intrinsic functions, they can't be imported. [`Function::isIntrinsic`](https://llvm.org/doxygen/classllvm_1_1Function.html#a900a32da3983469187b1848189681705)